<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Preface | GWAS Tutorial</title>
<meta name="author" content="Yen-Hsiang (Teddy) Huang">
<meta name="description" content="This GitHub page provides a hands-on tutorial on genome-wide association studies (GWAS) as part of Teddy’s and Lumi’s research program at Texas A&amp;M University under the supervision of Dr. Endang...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="Preface | GWAS Tutorial">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/TeddYenn/GWAS_RICE/preface.html">
<meta property="og:image" content="https://github.com/TeddYenn/GWAS_RICE/images/cover.png">
<meta property="og:description" content="This GitHub page provides a hands-on tutorial on genome-wide association studies (GWAS) as part of Teddy’s and Lumi’s research program at Texas A&amp;M University under the supervision of Dr. Endang...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Preface | GWAS Tutorial">
<meta name="twitter:description" content="This GitHub page provides a hands-on tutorial on genome-wide association studies (GWAS) as part of Teddy’s and Lumi’s research program at Texas A&amp;M University under the supervision of Dr. Endang...">
<meta name="twitter:image" content="https://github.com/TeddYenn/GWAS_RICE/images/cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Lato-0.4.9/font.css" rel="stylesheet">
<link href="libs/Roboto_Mono-0.4.9/font.css" rel="stylesheet">
<link href="libs/Montserrat-0.4.9/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.8.0/transition.js"></script><script src="libs/bs3compat-0.8.0/tabs.js"></script><script src="libs/bs3compat-0.8.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="For Anaerobic Germination in Rice">GWAS Tutorial</a>:
        <small class="text-muted">For Anaerobic Germination in Rice</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome to GWAS Tutorial</a></li>
<li><a class="active" href="preface.html">Preface</a></li>
<li><a class="" href="sec-genotype-data.html"><span class="header-section-number">1</span> Genotype Data</a></li>
<li><a class="" href="sec-phenotype-data.html"><span class="header-section-number">2</span> Phenotype Data</a></li>
<li><a class="" href="sec-gw.html"><span class="header-section-number">3</span> GWAS</a></li>
<li><a class="" href="sec-post-gwas.html"><span class="header-section-number">4</span> Post-GWAS</a></li>
<li><a class="" href="sec-extra.html"><span class="header-section-number">5</span> Extra Analysis and Visualization</a></li>
<li><a class="" href="sec-findings.html"><span class="header-section-number">6</span> Findings</a></li>
<li><a class="" href="sec-summary.html"><span class="header-section-number">7</span> Summary</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/TeddYenn/GWAS_RICE">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="preface" class="section level1 unnumbered">
<h1>Preface<a class="anchor" aria-label="anchor" href="#preface"><i class="fas fa-link"></i></a>
</h1>
<p>This GitHub page provides a hands-on tutorial on genome-wide association
studies (GWAS) as part of Teddy’s and Lumi’s research program at Texas
A&amp;M University under the supervision of Dr. Endang Septiningsih and Dr.
Michael Thomson.</p>
<p>The tutorial focuses on standard workflow, codes, and resources needed
to perform a GWAS in rice and is divided into seven sections. For
absolute beginners, we’ve also included an introductory section on
<a href="https://www.r-project.org/">R</a>,
<a href="https://posit.co/download/rstudio-desktop/">RStudio</a>, and
<a href="https://www.cog-genomics.org/plink/">PLINK</a>. <strong>-&gt; <a href="https://teddyenn.github.io/GWAS_RICE/preface.html#sec-tutorial-for-r-rstudio--plink">Click
Here!</a></strong></p>
<p>This book was written in <a href="http://www.rstudio.com/ide/">RStudio</a> using
<a href="http://bookdown.org/">bookdown</a>. The
<a href="https://teddyenn.github.io/GWAS_RICE/">website</a> is hosted via GitHub
under <a href="https://github.com/TeddYenn">TeddYenn’s repository.</a></p>
<p>If you have any questions or suggestions, feel free to reach out via
email at <a href="mailto:teddyenn2@gmail.com" class="email">teddyenn2@gmail.com</a>
(Teddy Huang).</p>
<div id="outline" class="section level2 unnumbered">
<h2>Outline<a class="anchor" aria-label="anchor" href="#outline"><i class="fas fa-link"></i></a>
</h2>
<p>We begin by processing genotypic and phenotypic datasets, followed by
conducting GWAS and post-GWAS analyses, and conclude with findings and
interpretations of the results.</p>
<ul>
<li>
<strong>Chapter</strong> <a href="sec-genotype-data.html#sec-genotype-data">1</a> downloads the SNP data from
SNP-Seek (3K RG), align the IDs, perform SNP QC, and generate the
final GWAS data.</li>
<li>
<strong>Chapter</strong> <a href="sec-phenotype-data.html#sec-phenotype-data">2</a> explores the phenotypic data
and generate the final GWAS data.</li>
<li>
<strong>Chapter</strong> <a href="sec-gw.html#sec-gw">3</a> conduct the genome-wide association study
(GWAS) to link genotypic variation with the phenotypic traits of
interest.</li>
<li>
<strong>Chapter</strong> <a href="sec-post-gwas.html#sec-post-gwas">4</a> delve into post-GWAS analyses that
further dissect and validate the significant associations identified
in the GWAS.</li>
<li>
<strong>Chapter</strong> <a href="sec-extra.html#sec-extra">5</a> present additional exploratory analyses
that extend beyond the primary GWAS objectives.</li>
</ul>
<p>We provide the detailed findings and make some narratives about the
results.</p>
<ul>
<li>
<strong>Chapter</strong> <a href="sec-findings.html#sec-findings">6</a> synthesize the key findings from the
genotypic, phenotypic, GWAS, and post-GWAS analyses.</li>
<li>
<strong>Chapter</strong> <a href="sec-summary.html#sec-summary">7</a> summarize the overall outcomes of the
study, emphasizing major insights gained and acknowledging any
limitations encountered.</li>
</ul>
</div>
<div id="sec-tutorial-for-r-rstudio--plink" class="section level2 unnumbered">
<h2>Tutorial for R, RStudio &amp; PLINK<a class="anchor" aria-label="anchor" href="#sec-tutorial-for-r-rstudio--plink"><i class="fas fa-link"></i></a>
</h2>
<details><summary><span style="color: white; background-color: grey; padding: 2px 4px; border-radius: 3px;"><strong>How to download the R &amp;
RStudio</strong></span>.
</summary><div class="rmdtip">
<p>
For TAMU HPRC users, you don’t need to download RStudio. There is the
built-in RStudio can be launch online on HPRC.
</p>
</div>
<p><strong>Step 1: Download and Install R (Prerequisite)</strong></p>
<ol style="list-style-type: decimal">
<li>
<p>Visit the official R Project website: <a href="https://www.r-project.org/" class="uri">https://www.r-project.org/</a>.</p>
<ul>
<li><p><strong>Windows</strong>: Click <strong>Download R for Windows</strong>, select “base,”
and download the latest version.</p></li>
<li><p><strong>MacOS</strong>: Click <strong>Download R for macOS</strong> and choose the correct
version for your system.</p></li>
</ul>
</li>
<li><p>Install R by double-clicking the downloaded installer and following
the on-screen instructions. Use the default options unless specific
needs arise.</p></li>
</ol>
<p><strong>Step 2: Download RStudio</strong></p>
<ol style="list-style-type: decimal">
<li><p>Go to the RStudio official download page:
<a href="https://posit.co/download/rstudio-desktop/" class="uri">https://posit.co/download/rstudio-desktop/</a>.</p></li>
<li><p>Click <strong>Download</strong> under “RStudio Desktop - Open Source License.”</p></li>
<li>
<p>Select the version suitable for your operating system:</p>
<ul>
<li><p><strong>Windows</strong>: <code>.exe</code> file</p></li>
<li><p><strong>MacOS</strong>: <code>.dmg</code> file</p></li>
</ul>
</li>
</ol>
<p><strong>Step 3: Install RStudio</strong></p>
<ol style="list-style-type: decimal">
<li>
<p>Locate the downloaded file and double-click it to start the
installation.</p>
<ul>
<li><p><strong>Windows</strong>: Run the <code>.exe</code> installer and follow the wizard
steps.</p></li>
<li><p><strong>MacOS</strong>: Drag the RStudio icon into the Applications folder.</p></li>
</ul>
</li>
<li><p>After installation, launch RStudio.</p></li>
</ol>
<p><strong>Step 4: Launch RStudio</strong></p>
<ol style="list-style-type: decimal">
<li><p>Open RStudio by clicking the shortcut created during installation or
searching for “RStudio” in your system’s application launcher.</p></li>
<li><p>RStudio will automatically detect your R installation and link to
it.</p></li>
</ol>
<p><strong>Step 5: Familiarize Yourself with RStudio Interface</strong></p>
<p>RStudio has four main panels:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Console (Bottom-left)</strong>: Where you run R commands.</p></li>
<li><p><strong>Source (Top-left)</strong>: For writing and editing scripts.</p></li>
<li><p><strong>Environment/History (Top-right)</strong>: Displays objects, variables,
and command history.</p></li>
<li><p><strong>Plots/Files/Help (Bottom-right)</strong>: Displays plots, files, and R
documentation.</p></li>
</ol>
<hr></details><details><summary><span style="color: white; background-color: grey; padding: 2px 4px; border-radius: 3px;"><strong>How to download the R
packages</strong></span>.
</summary><p><strong>Step 1: Understand R Packages</strong></p>
<p>R packages are collections of functions, data, and documentation that
extend R’s capabilities. These packages are hosted on CRAN,
Bioconductor, or GitHub.</p>
<p><strong>Step 2: Install CRAN Packages</strong></p>
<ol style="list-style-type: decimal">
<li>
<p><strong>Basic Installation</strong>: Use the <code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages()</a></code> function to
install packages from CRAN. For example: This command downloads and
installs the <code>ggplot2</code> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"ggplot2"</span><span class="op">)</span> </span></code></pre></div>
<p><strong>Load the Package</strong>: After installation, load the package using:
Once loaded, you can use its functions.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span> </span></code></pre></div>
</li>
</ol>
<p><strong>Step 3: Install Bioconductor Packages</strong></p>
<p>Bioconductor hosts many bioinformatics packages. To install, follow
these steps:</p>
<ol style="list-style-type: decimal">
<li>
<p><strong>Install BiocManager</strong>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span> </span></code></pre></div>
</li>
<li>
<p><strong>Install Bioinformatics Packages:</strong> Use <code><a href="https://bioconductor.github.io/BiocManager/reference/install.html">BiocManager::install()</a></code> to
install Bioconductor packages. Example: This installs the <code>ggtree</code>
package.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html">install</a></span><span class="op">(</span><span class="st">"ggtree"</span><span class="op">)</span> </span></code></pre></div>
</li>
<li>
<p><strong>Load the Package</strong>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://www.amazon.com/Integration-Manipulation-Visualization-Phylogenetic-Computational-ebook/dp/B0B5NLZR1Z/">ggtree</a></span><span class="op">)</span></span></code></pre></div>
</li>
</ol>
<p><strong>Step 4: Install GitHub Packages</strong></p>
<p>Some specialized packages are hosted on GitHub. To install these:</p>
<ol style="list-style-type: decimal">
<li>
<p><strong>Install <code>remotes</code> or <code>devtools</code></strong>:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span> </span></code></pre></div>
</li>
<li>
<p><strong>Install a Package from GitHub</strong>: Use <code><a href="https://remotes.r-lib.org/reference/install_github.html">remotes::install_github()</a></code>.
For example: This installs the <code>GAPIT3</code> package for GWAS analysis.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"jiabowang/GAPIT3"</span><span class="op">)</span>  </span></code></pre></div>
</li>
<li>
<p><strong>Load the Package</strong>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">GAPIT3</span><span class="op">)</span> </span></code></pre></div>
</li>
</ol>
<p><strong>Step 5: Update R Packages</strong></p>
<ol style="list-style-type: decimal">
<li>
<p>To update all installed packages:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/update.packages.html">update.packages</a></span><span class="op">(</span>ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </span></code></pre></div>
</li>
</ol>
<!-- --><ol start="2" style="list-style-type: decimal">
<li>
<p>For Bioconductor packages:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html">install</a></span><span class="op">(</span>update <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </span></code></pre></div>
</li>
</ol>
<p><strong>Step 6: Uninstall a Package (If Needed)</strong></p>
<p>If you need to remove a package:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/remove.packages.html">remove.packages</a></span><span class="op">(</span><span class="st">"ggplot2"</span><span class="op">)</span> </span></code></pre></div>
<p><strong>Summary of Installation Methods</strong></p>
<table class="table table-sm">
<thead><tr class="header">
<th>Source</th>
<th>Command Example</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>CRAN</td>
<td><code>install.packages("ggplot2")</code></td>
</tr>
<tr class="even">
<td>Bioconductor</td>
<td><code>BiocManager::install("ggtree")</code></td>
</tr>
<tr class="odd">
<td>GitHub</td>
<td><code>remotes::install_github("jiabowang/GAPIT3")</code></td>
</tr>
</tbody>
</table>
<hr></details><details><summary><span style="color: white; background-color: grey; padding: 2px 4px; border-radius: 3px;"><strong>How to download and use PLINK</strong> <strong>on TAMU
HPRC</strong></span>.
</summary><p><strong>Step 1: Download the PLINK File</strong></p>
<ol style="list-style-type: decimal">
<li><p>Visit the <a href="https://www.cog-genomics.org/plink/1.9/">PLINK website</a>.</p></li>
<li><p>Download the <strong>Linux 64-bit</strong> version, which is compatible with the
<a href="https://hprc.tamu.edu/kb/User-Guides/FASTER/Hardware/">FASTER</a> and
<a href="https://hprc.tamu.edu/kb/User-Guides/Grace/Hardware/">Grace</a>
system.</p></li>
</ol>
<p><strong>Step 2: Extract and Upload the PLINK File to TAMU HPRC</strong></p>
<ol style="list-style-type: decimal">
<li><p>Extract the downloaded PLINK file (e.g., <code>plink_linux_x86_64</code>).</p></li>
<li>
<p>Upload the extracted PLINK folder to TAMU HPRC via the file transfer
method you prefer. For example:</p>
<ul>
<li><p>Create an <code>Upload</code> folder in your home directory
(<code>/home/&lt;username&gt;/Upload</code>) or scratch
(<code>/scratch/user/&lt;username&gt;/Upload</code>).</p></li>
<li><p>Place the PLINK folder (<code>plink_linux_x86_64</code>) inside this
directory.</p></li>
</ul>
</li>
</ol>
<p><strong>Step 3: Open the Terminal on TAMU HPRC</strong></p>
<ol style="list-style-type: decimal">
<li><p>Log in to the TAMU HPRC portal.</p></li>
<li><p>Navigate to your working directory where the PLINK file was uploaded
(e.g. <code>/home/&lt;username&gt;/Upload/plink_linux_x86_64</code>).</p></li>
<li><p>Click <strong><code>&gt;_ Open in Terminal</code></strong> to access the terminal interface.</p></li>
</ol>
<p><strong>Step 4: Verify Your Current Directory</strong></p>
<ol style="list-style-type: decimal">
<li>
<p>In the terminal, run the command after $</p>
<pre><code>pwd </code></pre>
</li>
<li>
<p>Confirm that you are in the directory where the PLINK file is
located. For example:</p>
<pre><code> /home/&lt;username&gt;/Upload/plink_linux_x86_64 </code></pre>
</li>
</ol>
<p><strong>Step 5: Make the PLINK File Executable and Test It</strong></p>
<ul>
<li>
<p>In the terminal, navigate to the PLINK folder if not already there:</p>
<pre><code>cd /home/&lt;username&gt;/Upload/plink_linux_x86_6 </code></pre>
</li>
<li>
<p>Run the following commands:</p>
<pre><code>chmod +x plink</code></pre>
<pre><code>./plink --help</code></pre>
<ul>
<li>
<strong><code>chmod +x plink</code></strong>: Grants the PLINK file executable
permissions.</li>
<li>
<strong><code>./plink --help</code></strong>: Runs the PLINK executable and displays
help information to confirm proper setup.</li>
</ul>
</li>
</ul>
<p><strong>Step 6: Upload Your Data and Use PLINK</strong></p>
<ol style="list-style-type: decimal">
<li><p>Upload your data files (e.g., <code>.bed</code>, <code>.bim</code>, <code>.fam</code> files) into the
working directory (<code>/home/&lt;username&gt;/Upload/plink_linux_x86_64</code>).</p></li>
<li><p>Use PLINK commands to process your data.</p></li>
</ol>
<p><strong>Example: Converting <code>.bfile</code> to <code>.vcf</code></strong></p>
<ol style="list-style-type: decimal">
<li>
<p>Assuming your <code>.bed</code>, <code>.bim</code>, and <code>.fam</code> files are named <code>data.bed</code>,
<code>data.bim</code>, and <code>data.fam</code>, use the following command:</p>
<pre><code>./plink --bfile data --recode vcf --out data </code></pre>
</li>
</ol>
<!-- --><ol start="2" style="list-style-type: decimal">
<li>This will generate a VCF file named <code>data.vcf</code> in your current
directory.</li>
</ol>
<hr></details>
</div>
<div id="acknowledgements" class="section level2 unnumbered">
<h2>Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"><i class="fas fa-link"></i></a>
</h2>
<p>We are deeply grateful for the contributions, support, and perspectives
of individuals and organizations that have helped move this project
forward.</p>
<p>We would like to extend our sincere thanks to Dr. Endang Septiningsih
and Dr. Michael Thomson for hosting us at Texas A&amp;M University. We also
greatly appreciate the support provided by the International Agriculture
Center, National Chung Hsing University, and the Higher Education Sprout
Project, Ministry of Education, Taiwan, for offering opportunities and
partial financial support for this research program.</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="index.html">Welcome to GWAS Tutorial</a></div>
<div class="next"><a href="sec-genotype-data.html"><span class="header-section-number">1</span> Genotype Data</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#preface">Preface</a></li>
<li><a class="nav-link" href="#outline">Outline</a></li>
<li><a class="nav-link" href="#sec-tutorial-for-r-rstudio--plink">Tutorial for R, RStudio &amp; PLINK</a></li>
<li><a class="nav-link" href="#acknowledgements">Acknowledgements</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/TeddYenn/GWAS_RICE/blob/master/index.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/TeddYenn/GWAS_RICE/edit/master/index.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>GWAS Tutorial</strong>: For Anaerobic Germination in Rice" was written by Yen-Hsiang (Teddy) Huang. It was last built on 2025-01-28.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
