<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>3 GWAS | GWAS Tutorial</title>
<meta name="author" content="Yen-Hsiang (Teddy) Huang">
<meta name="description" content="Conduct GWAS Analysis and Collect All Results Preview the GWAS Results  3.1 GWAS Analysis We used the GAPIT R package, which offers multiple GWAS models, to conduct GWAS analyses. Four separate...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="3 GWAS | GWAS Tutorial">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/TeddYenn/GWAS_RICE/sec-gw.html">
<meta property="og:image" content="https://github.com/TeddYenn/GWAS_RICE/images/cover.png">
<meta property="og:description" content="Conduct GWAS Analysis and Collect All Results Preview the GWAS Results  3.1 GWAS Analysis We used the GAPIT R package, which offers multiple GWAS models, to conduct GWAS analyses. Four separate...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="3 GWAS | GWAS Tutorial">
<meta name="twitter:description" content="Conduct GWAS Analysis and Collect All Results Preview the GWAS Results  3.1 GWAS Analysis We used the GAPIT R package, which offers multiple GWAS models, to conduct GWAS analyses. Four separate...">
<meta name="twitter:image" content="https://github.com/TeddYenn/GWAS_RICE/images/cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Lato-0.4.9/font.css" rel="stylesheet">
<link href="libs/Roboto_Mono-0.4.9/font.css" rel="stylesheet">
<link href="libs/Montserrat-0.4.9/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.8.0/transition.js"></script><script src="libs/bs3compat-0.8.0/tabs.js"></script><script src="libs/bs3compat-0.8.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="For Anaerobic Germination in Rice">GWAS Tutorial</a>:
        <small class="text-muted">For Anaerobic Germination in Rice</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome to GWAS Tutorial</a></li>
<li><a class="" href="preface.html">Preface</a></li>
<li><a class="" href="sec-genotype-data.html"><span class="header-section-number">1</span> Genotype Data</a></li>
<li><a class="" href="sec-phenotype-data.html"><span class="header-section-number">2</span> Phenotype Data</a></li>
<li><a class="active" href="sec-gw.html"><span class="header-section-number">3</span> GWAS</a></li>
<li><a class="" href="sec-post-gwas.html"><span class="header-section-number">4</span> Post-GWAS</a></li>
<li><a class="" href="sec-extra.html"><span class="header-section-number">5</span> Extra Analysis and Visualization</a></li>
<li><a class="" href="sec-findings.html"><span class="header-section-number">6</span> Findings</a></li>
<li><a class="" href="sec-summary.html"><span class="header-section-number">7</span> Summary</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/TeddYenn/GWAS_RICE">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="sec-gw" class="section level1" number="3">
<h1>
<span class="header-section-number">3</span> GWAS<a class="anchor" aria-label="anchor" href="#sec-gw"><i class="fas fa-link"></i></a>
</h1>
<ol style="list-style-type: decimal">
<li><p><strong>Conduct GWAS Analysis and Collect All Results</strong></p></li>
<li><p><strong>Preview the GWAS Results</strong></p></li>
</ol>
<div id="gwas-analysis" class="section level2" number="3.1">
<h2>
<span class="header-section-number">3.1</span> <strong>GWAS Analysis</strong><a class="anchor" aria-label="anchor" href="#gwas-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>We used the <a href="https://zzlab.net/GAPIT/gapit_help_document.pdf"><strong>GAPIT</strong></a>
R package, which offers multiple GWAS models, to conduct GWAS analyses.</p>
<p>Four separate GWAS results were stored in files GWAS1 through GWAS4. In
each analysis, we applied five models simultaneously: GLM, MLM, MLMM,
FarmCPU, and BLINK.</p>
<div class="rmdnote">
<p>
Genotypic dataset (Geno.rds): 327 samples and 1.9 million SNPs.
Phenotypic dataset (Pheno.rds): 327 samples and 3 traits.
</p>
</div>
<details><summary>
➡️ <strong>Working on R (RStudio)</strong>
</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##### R CODES #####</span></span>
<span></span>
<span><span class="co"># Set the working directory for initial setup</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"...change..."</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load genotype and phenotype data</span></span>
<span><span class="va">Geno</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"Geno.rds"</span><span class="op">)</span></span>
<span><span class="va">Pheno</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"Pheno.rds"</span><span class="op">)</span> </span>
<span><span class="co"># 4 traits: Control_AG_14DAS; Control_AG_7DAS; AG_14DAS; AG_7DAS</span></span>
<span></span>
<span><span class="co"># Load the GAPIT library for GWAS analysis</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">GAPIT</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set the working directory for GWAS_PC2 analysis</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"...change.../GWAS_PC2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform GWAS analysis using different models and 2 principal components</span></span>
<span><span class="va">GAPIT</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GAPIT/man/GAPIT.html">GAPIT</a></span><span class="op">(</span></span>
<span>  Y <span class="op">=</span> <span class="va">Pheno</span>,</span>
<span>  G <span class="op">=</span> <span class="va">Geno</span>,</span>
<span>  model <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GLM"</span>, <span class="st">"MLM"</span>, <span class="st">"MLMM"</span>, <span class="st">"FarmCPU"</span>, <span class="st">"Blink"</span><span class="op">)</span>,</span>
<span>  PCA.total <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  Multiple_analysis <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set the working directory for GWAS_PC3 analysis</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"...change.../GWAS_PC3"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform GWAS analysis using different models and 3 principal components</span></span>
<span><span class="va">GAPIT</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GAPIT/man/GAPIT.html">GAPIT</a></span><span class="op">(</span></span>
<span>  Y <span class="op">=</span> <span class="va">Pheno</span>,</span>
<span>  G <span class="op">=</span> <span class="va">Geno</span>,</span>
<span>  model <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GLM"</span>, <span class="st">"MLM"</span>, <span class="st">"MLMM"</span>, <span class="st">"FarmCPU"</span>, <span class="st">"Blink"</span><span class="op">)</span>,</span>
<span>  PCA.total <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  Multiple_analysis <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set the working directory for GWAS_PC4 analysis</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"...change.../GWAS_PC4"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform GWAS analysis using different models and 4 principal components</span></span>
<span><span class="va">GAPIT</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GAPIT/man/GAPIT.html">GAPIT</a></span><span class="op">(</span></span>
<span>  Y <span class="op">=</span> <span class="va">Pheno</span>,</span>
<span>  G <span class="op">=</span> <span class="va">Geno</span>,</span>
<span>  model <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GLM"</span>, <span class="st">"MLM"</span>, <span class="st">"MLMM"</span>, <span class="st">"FarmCPU"</span>, <span class="st">"Blink"</span><span class="op">)</span>,</span>
<span>  PCA.total <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  Multiple_analysis <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set the working directory for GWAS_PC5 analysis</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"...change.../GWAS_PC5"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform GWAS analysis using different models and 5 principal components</span></span>
<span><span class="va">GAPIT</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GAPIT/man/GAPIT.html">GAPIT</a></span><span class="op">(</span></span>
<span>  Y <span class="op">=</span> <span class="va">Pheno</span>,</span>
<span>  G <span class="op">=</span> <span class="va">Geno</span>,</span>
<span>  model <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GLM"</span>, <span class="st">"MLM"</span>, <span class="st">"MLMM"</span>, <span class="st">"FarmCPU"</span>, <span class="st">"Blink"</span><span class="op">)</span>,</span>
<span>  PCA.total <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  Multiple_analysis <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</details>
</div>
<div id="preview-the-gwas-results" class="section level2" number="3.2">
<h2>
<span class="header-section-number">3.2</span> <strong>Preview the GWAS Results</strong><a class="anchor" aria-label="anchor" href="#preview-the-gwas-results"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li><p><strong>GAPIT.Association.Filter_GWAS_results.csv</strong>: All of the
significant SNP markers from the selected models and phenotypic
traits.</p></li>
<li><p><strong>GAPIT.Association.Manhattan_Geno.Model.Trait.pdf</strong>: Genome-wide
Manhattan plots for each model and specific phenotypic trait.</p></li>
</ul>
<p>Here are the results: We only included significant SNP markers from
<strong>FarmCPU</strong> and <strong>BLINK</strong> (the newest models) that were associated with
<strong>Logit_Control_AG_14DAS</strong> (log-transformed data of the difference in
seed germination rates between control and treatment groups at 14 days).</p>
<div class="inline-table"><table class="table table-sm">
<thead><tr class="header">
<th>File name</th>
<th><em>GWAS_PC1</em></th>
<th><em>GWAS_PC2</em></th>
<th><em>GWAS_PC3</em></th>
<th><em>GWAS_PC4</em></th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Number of PCs</td>
<td>2</td>
<td>3</td>
<td>4</td>
<td>5</td>
</tr>
<tr class="even">
<td>FarmCPU</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>BLINK</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table></div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="sec-phenotype-data.html"><span class="header-section-number">2</span> Phenotype Data</a></div>
<div class="next"><a href="sec-post-gwas.html"><span class="header-section-number">4</span> Post-GWAS</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#sec-gw"><span class="header-section-number">3</span> GWAS</a></li>
<li><a class="nav-link" href="#gwas-analysis"><span class="header-section-number">3.1</span> GWAS Analysis</a></li>
<li><a class="nav-link" href="#preview-the-gwas-results"><span class="header-section-number">3.2</span> Preview the GWAS Results</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/TeddYenn/GWAS_RICE/blob/master/03-gwas.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/TeddYenn/GWAS_RICE/edit/master/03-gwas.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>GWAS Tutorial</strong>: For Anaerobic Germination in Rice" was written by Yen-Hsiang (Teddy) Huang. It was last built on 2025-01-23.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
